Sources/VLCRendererDiscoverer.m

```diff
@@ -153,16 +154,37 @@ - (void)dealloc
     return [list copy];
 }
 
+- (VLCRendererItem *)discoveredItemsContainsItem:(libvlc_renderer_item_t *)item
+{
+    for (VLCRendererItem *tmpItem in _items) {
+        if (tmpItem.rendererItem == item) {
+            return tmpItem;
+        }
+    }
+    return nil;
+}
+
 #pragma mark - Handling libvlc event callbacks
 
-- (void)itemAdded:(VLCRendererItem *)item
+- (void)itemAdded:(NSValue *)item
 {
-    [_delegate rendererDiscovererItemAdded:self item:item];
+    libvlc_renderer_item_t *pointerValue = item.pointerValue;
+    VLCRendererItem *tmp = [self discoveredItemsContainsItem:pointerValue];
+
+    if (!tmp) {
+        tmp = [[VLCRendererItem alloc] initWithCItem: pointerValue];
+        [_items addObject:tmp];
+    }
+    [_delegate rendererDiscovererItemAdded:self item:tmp];
 }
 
-- (void)itemDeleted:(VLCRendererItem *)item
+- (void)itemDeleted:(NSValue *)item
 {
-    [_delegate rendererDiscovererItemDeleted:self item:item];
+    libvlc_renderer_item_t *pointerValue = item.pointerValue;
+    VLCRendererItem *tmp = [self discoveredItemsContainsItem:pointerValue];
+
+    [_items removeObject:tmp];
+    [_delegate rendererDiscovererItemDeleted:self item:tmp];
```

Comment:
This would turn into : 
 for (VLCRendererItem *tmpItem in _items) {
         if (tmpItem.rendererItem == item.pointerValue) {
             [_items removeObject: tmpItem];
             [_delegate rendererDiscovererItemDeleted:self item: tmpItem];
             return; 
         }
     }

Though double check that you don't get an issue with modified while being iterated